rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1635,119 +1659,133 @@ private Optional<Federation> getFederationFromPublicKey(BtcECKey federatorPublic
+            try {
+                decodedSignatures.add(BtcECKey.ECDSASignature.decodeFromDER(signature));
+            } catch (RuntimeException e) {
+                int index = signatures.indexOf(signature);
+                logger.warn("Malformed signature for input {} : {}", index, Bytes.of(signature));
+                throw new SignatureException();
+            }
+        }
+        return decodedSignatures;
+    }
 
-        // All signatures are correct. Proceed to signing
-        for (int i = 0; i < numInputs; i++) {
-            Sha256Hash sighash = sighashes.get(i);
+    private boolean sign(
+        BtcECKey federatorBtcPublicKey,
+        List<TransactionSignature> txSigs,
+        List<Sha256Hash> sigHashes,
+        Keccak256 rskTxHash,
+        BtcTransaction btcTx) {
+
+        boolean signed = false;
+        for (int i = 0; i < sigHashes.size(); i++) {
+            Sha256Hash sighash = sigHashes.get(i);
             TransactionInput input = btcTx.getInput(i);
             Script inputScript = input.getScriptSig();
 
-            boolean alreadySignedByThisFederator = BridgeUtils.isInputSignedByThisFederator(
-                    federatorBtcPublicKey,
-                    sighash,
-                    input);
+            boolean alreadySignedByThisFederator =
+                BridgeUtils.isInputSignedByThisFederator(federatorBtcPublicKey, sighash, input);
 
             // Sign the input if it wasn't already
-            if (!alreadySignedByThisFederator) {
-                try {
-                    int sigIndex = inputScript.getSigInsertionIndex(sighash, federatorBtcPublicKey);
-                    inputScript = ScriptBuilder.updateScriptWithSignature(inputScript, txSigs.get(i).encodeToBitcoin(), sigIndex, 1, 1);
-                    input.setScriptSig(inputScript);
-                    logger.debug("Tx input {} for tx {} signed.", i, new Keccak256(rskTxHash));
-                    signed = true;
-                } catch (IllegalStateException e) {
-                    Federation retiringFederation = getRetiringFederation();
-                    if (getActiveFederation().hasBtcPublicKey(federatorBtcPublicKey)) {
-                        logger.debug("A member of the active federation is trying to sign a tx of the retiring one");
-                        return;
-                    } else if (retiringFederation != null && retiringFederation.hasBtcPublicKey(federatorBtcPublicKey)) {
-                        logger.debug("A member of the retiring federation is trying to sign a tx of the active one");
-                        return;
-                    }
-                    throw e;
-                }
-            } else {
-                logger.warn("Input {} of tx {} already signed by this federator.", i, new Keccak256(rskTxHash));
+            if (alreadySignedByThisFederator) {
+                logger.warn("Input {} of tx {} already signed by this federator.", i, rskTxHash);
```

Comment:
Same here, logging the rsk tx has is probably not what was intended. I'd check this in the remaining places